src/Maps/components/LeafletMap.ts

```diff
@@ -0,0 +1,192 @@
+        });
+    }
+
+    private setDefaultCenter = (props: LeafletMapProps) => {
+        const { defaultCenterLatitude, defaultCenterLongitude, fetchingData } = props;
+        if (defaultCenterLatitude && defaultCenterLongitude) {
+            this.setState({
+                center: {
+                    lat: Number(defaultCenterLatitude),
+                    lng: Number(props.defaultCenterLongitude)
+                }
+            });
+        } else if (!fetchingData) {
+            this.renderMarkers(props.allLocations);
+        }
+    }
+
+    private renderMarkers = (locations?: Location[]) => {
+        this.markerGroup.clearLayers();
+        if (locations && locations.length) {
+            locations.forEach(location =>
+                this.createMarker(location)
+                    .then(marker =>
+                        this.markerGroup.addLayer(marker
+                            .on("click", event => {
+                                if (this.props.onClickMarker && location.locationAttr) {
+                                    this.props.onClickMarker(event, location.locationAttr);
+                                }
+                            }
+                        )))
+                    .then(layer =>
+                        this.map
+                            ? this.map.addLayer(layer)
+                            : undefined)
+                    .catch(reason =>
+                        this.setState({ alertMessage: `${reason}` })));
+            this.setBounds();
+        } else if (this.map) {
+            this.map.removeLayer(this.markerGroup);
+        }
+    }
+
+    private setBounds = () => {
+        setTimeout(() => {
+            if (this.map) {
+                try {
+                    this.map.fitBounds(this.markerGroup.getBounds(), { animate: false }).invalidateSize();
+                    if (!this.props.autoZoom) {
+                        this.map.setZoom(this.props.zoomLevel, { animate: false });
+                    }
+                } catch (error) {
+                    this.setState({ alertMessage: `Failed due to ${error.message}` });
+                }
+            }
+        }, 0);
+    }
+
+    private createMarker = (location: Location): Promise<Marker> =>
+        new Promise((resolve, reject) => {
+            const { latitude, longitude, url } = location;
+            if (location) {
+                if (url) {
+                    resolve(
+                        new Marker([
+                            Number(latitude),
+                            Number(longitude)
+                        ]).setIcon(icon({
+                            iconUrl: url,
+                            iconSize: [ 38, 95 ],
+                            iconAnchor: [ 22, 94 ],
+                            className: "marker"
+                        }))
+                    );
+                } else {
+                    resolve(new Marker([
+                        Number(latitude),
```

Comment:
Inline... Code lines are precious... Don't be wasting them when you can avoid them. Not just for your sake, but also for the sake of your reviewers... we don't like looking through a million lines of code when they could be 100.

Reply:
done